[bug] panic in RetrieveRoleInfo after authorization when account retrieval fails due to ResultSize > 100

**Describe the bug**
After successful AWS IAM Identity Center device authorization, go-aws-sso fails while retrieving account information. The ListAccounts request returns HTTP 400 because the requested result size is outside the AWS valid range of 1-100.

**To Reproduce**
Steps to reproduce the behavior:

Run: `go-aws-sso`
Complete the AWS device authorization flow.
After authorization succeeds, account retrieval fails.
Modifying the code to log the upstream error show: InvalidRequestException: ResultSize parameter must be within the valid limits of: 1-100.

**Expected behavior**
After authorization, go-aws-sso should list available AWS accounts, allow account/role selection, and write or refresh credentials successfully.

**Error Output**

```
2026-05-15 09:02:44 INFO Still waiting for authorization...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102e05c3c]

goroutine 1 [running]:
github.com/theurichde/go-aws-sso/internal.RetrieveRoleInfo(0x0, {{0xc279ee7ee62cfeb8, 0x19ed32eef957, 0x103535ae0}, {0x14000524000, 0xe8}, {0x140000aa4e0, 0x22}, {0x140004a2000, 0x6c3}, ...}, ...)
	github.com/theurichde/go-aws-sso/internal/aws.go:15 +0x9c
main.start({0x1031a1e80, 0x1400000e260}, {0x1031a2398?, 0x1400000e280?}, 0x18?, {0x10319ecb8, 0x10356ae40})
	github.com/theurichde/go-aws-sso/cmd/go-aws-sso/main.go:194 +0x28c
main.main.func4(0x1400001ee40)
	github.com/theurichde/go-aws-sso/cmd/go-aws-sso/main.go:153 +0x140
github.com/urfave/cli/v2.(*Command).Run(0x140003a4840, 0x1400001ee40, {0x1400001e080, 0x4, 0x4})
	github.com/urfave/cli/v2@v2.25.0/command.go:273 +0x75c
github.com/urfave/cli/v2.(*App).RunContext(0x140003a6000, {0x10319f7a8?, 0x140000280c8}, {0x1400001e080, 0x4, 0x4})
	github.com/urfave/cli/v2@v2.25.0/app.go:332 +0x604
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.25.0/app.go:309
main.main()
	github.com/theurichde/go-aws-sso/cmd/go-aws-sso/main.go:162 +0xd44
```

--- a/cmd/go-aws-sso/main.go
+++ b/cmd/go-aws-sso/main.go
@@ -204,11 +204,18 @@ func start(oidcClient ssooidciface.SSOOIDCAPI, ssoClient ssoiface.SSOAPI, contex
 	clientInformation := ProcessClientInformation(oidcClient, startUrl)
 
 	accountInfo, awsErr := RetrieveAccountInfo(clientInformation, ssoClient, promptSelector)
+	if awsErr != nil {
+		fmt.Println(awsErr)
+	}
 	if awsErr != nil && awsErr.StatusCode() == 401 { // unauthorized
 		clientInformation, accountInfo = retryWithNewClientCreds(oidcClient, ssoClient, startUrl, promptSelector)
 	}
 
-	roleInfo := RetrieveRoleInfo(accountInfo, clientInformation, ssoClient, promptSelector)
+	roleInfo, roleErr := RetrieveRoleInfo(accountInfo, clientInformation, ssoClient, promptSelector)
+	if roleErr != nil {
+		fmt.Println(roleErr)
+		check(roleErr)
+	}
 	SaveUsageInformation(accountInfo, roleInfo)
 
 	rci := &sso.GetRoleCredentialsInput{AccountId: accountInfo.AccountId, RoleName: roleInfo.RoleName, AccessToken: &clientInformation.AccessToken}

--- a/cmd/go-aws-sso/main.go
+++ b/cmd/go-aws-sso/main.go
@@ -205,15 +205,14 @@ func start(oidcClient ssooidciface.SSOOIDCAPI, ssoClient ssoiface.SSOAPI, contex
 
 	accountInfo, awsErr := RetrieveAccountInfo(clientInformation, ssoClient, promptSelector)
 	if awsErr != nil {
-		fmt.Println(awsErr)
+		if awsErr.StatusCode() == 401 { // unauthorized
+			clientInformation, accountInfo = retryWithNewClientCreds(oidcClient, ssoClient, startUrl, promptSelector)
+		} else {
+			check(awsErr)
+		}
 	}
-	if awsErr != nil && awsErr.StatusCode() == 401 { // unauthorized
-		clientInformation, accountInfo = retryWithNewClientCreds(oidcClient, ssoClient, startUrl, promptSelector)
-	}
-
 	roleInfo, roleErr := RetrieveRoleInfo(accountInfo, clientInformation, ssoClient, promptSelector)
 	if roleErr != nil {
-		fmt.Println(roleErr)
 		check(roleErr)
 	}
 	SaveUsageInformation(accountInfo, roleInfo)

--- a/internal/aws.go
+++ b/internal/aws.go
@@ -34,17 +34,28 @@ func RetrieveRoleInfo(accountInfo *sso.AccountInfo, clientInformation ClientInfo
 }
 
 func RetrieveAccountInfo(clientInformation ClientInformation, ssoClient ssoiface.SSOAPI, selector Prompt) (*sso.AccountInfo, awserr.RequestFailure) {
-	var maxSize int64 = 1000 // default is 20, but sometimes you have more accounts available ;-)
+	var maxSize int64 = 100 // AWS SSO API limit is 100
 	lai := sso.ListAccountsInput{AccessToken: &clientInformation.AccessToken, MaxResults: &maxSize}
-	accounts, err := ssoClient.ListAccounts(&lai)
-	if err != nil {
-		if awsError, ok := err.(awserr.RequestFailure); ok {
-			return nil, awsError
+	
+	var allAccounts []*sso.AccountInfo
+	for {
+		accounts, err := ssoClient.ListAccounts(&lai)
+		if err != nil {
+			if awsError, ok := err.(awserr.RequestFailure); ok {
+				return nil, awsError
+			}
+			check(err)
 		}
+		
+		allAccounts = append(allAccounts, accounts.AccountList...)
+		
+		if accounts.NextToken == nil {
+			break
+		}
+		lai.NextToken = accounts.NextToken
 	}
-	check(err)
 
-	sortedAccounts := sortAccounts(accounts.AccountList)
+	sortedAccounts := sortAccounts(allAccounts)
 
 	var accountsToSelect []string
 	linePrefix := "#"

--- a/internal/aws.go
+++ b/internal/aws.go
@@ -11,26 +11,44 @@ import (
 	"strconv"
 )
 
-func RetrieveRoleInfo(accountInfo *sso.AccountInfo, clientInformation ClientInformation, ssoClient ssoiface.SSOAPI, selector Prompt) *sso.RoleInfo {
-	lari := &sso.ListAccountRolesInput{AccountId: accountInfo.AccountId, AccessToken: &clientInformation.AccessToken}
-	roles, _ := ssoClient.ListAccountRoles(lari)
+func RetrieveRoleInfo(accountInfo *sso.AccountInfo, clientInformation ClientInformation, ssoClient ssoiface.SSOAPI, selector Prompt) (*sso.RoleInfo, awserr.RequestFailure) {
+	var maxSize int64 = 100 // AWS SSO API limit is 100
+	lari := &sso.ListAccountRolesInput{AccountId: accountInfo.AccountId, AccessToken: &clientInformation.AccessToken, MaxResults: &maxSize}
 
-	if len(roles.RoleList) == 1 {
-		zap.S().Infof("Only one role available. Selected role: %s\n", *roles.RoleList[0].RoleName)
-		return roles.RoleList[0]
+	var allRoles []*sso.RoleInfo
+	for {
+		roles, err := ssoClient.ListAccountRoles(lari)
+		if err != nil {
+			if awsError, ok := err.(awserr.RequestFailure); ok {
+				return nil, awsError
+			}
+			check(err)
+		}
+
+		allRoles = append(allRoles, roles.RoleList...)
+
+		if roles.NextToken == nil {
+			break
+		}
+		lari.NextToken = roles.NextToken
+	}
+
+	if len(allRoles) == 1 {
+		zap.S().Infof("Only one role available. Selected role: %s\n", *allRoles[0].RoleName)
+		return allRoles[0], nil
 	}
 
 	var rolesToSelect []string
 	linePrefix := "#"
 
-	for i, info := range roles.RoleList {
+	for i, info := range allRoles {
 		rolesToSelect = append(rolesToSelect, linePrefix+strconv.Itoa(i)+" "+*info.RoleName)
 	}
 
 	label := "Select your role - Hint: fuzzy search supported. To choose one role directly just enter #{Int}"
 	indexChoice, _ := selector.Select(label, rolesToSelect, fuzzySearchWithPrefixAnchor(rolesToSelect, linePrefix))
-	roleInfo := roles.RoleList[indexChoice]
-	return roleInfo
+	roleInfo := allRoles[indexChoice]
+	return roleInfo, nil
 }
 
 func RetrieveAccountInfo(clientInformation ClientInformation, ssoClient ssoiface.SSOAPI, selector Prompt) (*sso.AccountInfo, awserr.RequestFailure) {

--- a/internal/refresh.go
+++ b/internal/refresh.go
@@ -38,8 +38,12 @@ func RefreshCredentials(oidcClient ssooidciface.SSOOIDCAPI, ssoClient ssoiface.S
 		if strings.Contains(err.Error(), "no such file") {
 			zap.S().Info("Nothing to refresh yet")
 			accountInfo, awsErr := RetrieveAccountInfo(clientInformation, ssoClient, Prompter{})
-			if awsErr != nil && awsErr.StatusCode() == 401 { // unauthorized
-				clientInformation, accountInfo = retryWithNewClientCreds(oidcClient, ssoClient, startUrl)
+			if awsErr != nil {
+				if awsErr.StatusCode() == 401 { // unauthorized
+					clientInformation, accountInfo = retryWithNewClientCreds(oidcClient, ssoClient, startUrl)
+				} else {
+					check(awsErr)
+				}
 			}
 			roleInfo, roleErr := RetrieveRoleInfo(accountInfo, clientInformation, ssoClient, Prompter{})
 			check(roleErr)

--- a/internal/refresh.go
+++ b/internal/refresh.go
@@ -41,7 +41,8 @@ func RefreshCredentials(oidcClient ssooidciface.SSOOIDCAPI, ssoClient ssoiface.S
 			if awsErr != nil && awsErr.StatusCode() == 401 { // unauthorized
 				clientInformation, accountInfo = retryWithNewClientCreds(oidcClient, ssoClient, startUrl)
 			}
-			roleInfo := RetrieveRoleInfo(accountInfo, clientInformation, ssoClient, Prompter{})
+			roleInfo, roleErr := RetrieveRoleInfo(accountInfo, clientInformation, ssoClient, Prompter{})
+			check(roleErr)
 			roleName = roleInfo.RoleName
 			accountId = accountInfo.AccountId
 			SaveUsageInformation(accountInfo, roleInfo)

